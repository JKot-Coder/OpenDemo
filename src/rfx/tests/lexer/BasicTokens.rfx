//LEXER_TEST:

Инвалид
identifier
123
123.36
"String literal"
'C'
// Line comment
/* Block
   Comment */

; , . .. ...
{}[]()
= + - * / % ! ~ << >> == != > < >= <= && || & | ^ ++ --
+= -= *= /= %= <<= >>= &= |= ^=
? : -> @ $ # ## ::