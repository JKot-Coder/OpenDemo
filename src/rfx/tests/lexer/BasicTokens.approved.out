LexerOutput:[
	{"Type":"line comment", "Content":"\/\/LEXER_TEST:", "Line":1, "Column":1},
	{"Type":"end of line", "Content":"\r\n", "Line":1, "Column":14},
	{"Type":"end of line", "Content":"\r\n", "Line":2, "Column":1},
	{"Type":"invalid character", "Content":"Инвалид", "Line":3, "Column":1},
	{"Type":"end of line", "Content":"\r\n", "Line":3, "Column":8},
	{"Type":"identifier", "Content":"identifier", "Line":4, "Column":1},
	{"Type":"end of line", "Content":"\r\n", "Line":4, "Column":11},
	{"Type":"integer literal", "Content":"123", "Line":5, "Column":1},
	{"Type":"end of line", "Content":"\r\n", "Line":5, "Column":4},
	{"Type":"floating-point literal", "Content":"123.36", "Line":6, "Column":1},
	{"Type":"end of line", "Content":"\r\n", "Line":6, "Column":7},
	{"Type":"string literal", "Content":"\"String literal\"", "Line":7, "Column":1},
	{"Type":"end of line", "Content":"\r\n", "Line":7, "Column":17},
	{"Type":"character literal", "Content":"'C'", "Line":8, "Column":1},
	{"Type":"end of line", "Content":"\r\n", "Line":8, "Column":4},
	{"Type":"line comment", "Content":"\/\/ Line comment", "Line":9, "Column":1},
	{"Type":"end of line", "Content":"\r\n", "Line":9, "Column":16},
	{"Type":"block comment", "Content":"\/* Block\r\n   Comment *\/", "Line":10, "Column":1},
	{"Type":"end of line", "Content":"\r\n", "Line":11, "Column":14},
	{"Type":"end of line", "Content":"\r\n", "Line":12, "Column":1},
	{"Type":"';'", "Content":";", "Line":13, "Column":1},
	{"Type":"whitespace", "Content":" ", "Line":13, "Column":2},
	{"Type":"','", "Content":",", "Line":13, "Column":3},
	{"Type":"whitespace", "Content":" ", "Line":13, "Column":4},
	{"Type":"'.'", "Content":".", "Line":13, "Column":5},
	{"Type":"whitespace", "Content":" ", "Line":13, "Column":6},
	{"Type":"'..'", "Content":"..", "Line":13, "Column":7},
	{"Type":"whitespace", "Content":" ", "Line":13, "Column":9},
	{"Type":"'...'", "Content":"...", "Line":13, "Column":10},
	{"Type":"end of line", "Content":"\r\n", "Line":13, "Column":13},
	{"Type":"'{'", "Content":"{", "Line":14, "Column":1},
	{"Type":"'}'", "Content":"}", "Line":14, "Column":2},
	{"Type":"'['", "Content":"[", "Line":14, "Column":3},
	{"Type":"']'", "Content":"]", "Line":14, "Column":4},
	{"Type":"'('", "Content":"(", "Line":14, "Column":5},
	{"Type":"')'", "Content":")", "Line":14, "Column":6},
	{"Type":"end of line", "Content":"\r\n", "Line":14, "Column":7},
	{"Type":"'='", "Content":"=", "Line":15, "Column":1},
	{"Type":"whitespace", "Content":" ", "Line":15, "Column":2},
	{"Type":"'+'", "Content":"+", "Line":15, "Column":3},
	{"Type":"whitespace", "Content":" ", "Line":15, "Column":4},
	{"Type":"'-'", "Content":"-", "Line":15, "Column":5},
	{"Type":"whitespace", "Content":" ", "Line":15, "Column":6},
	{"Type":"'*'", "Content":"*", "Line":15, "Column":7},
	{"Type":"whitespace", "Content":" ", "Line":15, "Column":8},
	{"Type":"'/'", "Content":"\/", "Line":15, "Column":9},
	{"Type":"whitespace", "Content":" ", "Line":15, "Column":10},
	{"Type":"'%'", "Content":"%", "Line":15, "Column":11},
	{"Type":"whitespace", "Content":" ", "Line":15, "Column":12},
	{"Type":"'!'", "Content":"!", "Line":15, "Column":13},
	{"Type":"whitespace", "Content":" ", "Line":15, "Column":14},
	{"Type":"'~'", "Content":"~", "Line":15, "Column":15},
	{"Type":"whitespace", "Content":" ", "Line":15, "Column":16},
	{"Type":"'<<'", "Content":"<<", "Line":15, "Column":17},
	{"Type":"whitespace", "Content":" ", "Line":15, "Column":19},
	{"Type":"'>>'", "Content":">>", "Line":15, "Column":20},
	{"Type":"whitespace", "Content":" ", "Line":15, "Column":22},
	{"Type":"'=='", "Content":"==", "Line":15, "Column":23},
	{"Type":"whitespace", "Content":" ", "Line":15, "Column":25},
	{"Type":"'!='", "Content":"!=", "Line":15, "Column":26},
	{"Type":"whitespace", "Content":" ", "Line":15, "Column":28},
	{"Type":"'>'", "Content":">", "Line":15, "Column":29},
	{"Type":"whitespace", "Content":" ", "Line":15, "Column":30},
	{"Type":"'<'", "Content":"<", "Line":15, "Column":31},
	{"Type":"whitespace", "Content":" ", "Line":15, "Column":32},
	{"Type":"'>='", "Content":">=", "Line":15, "Column":33},
	{"Type":"whitespace", "Content":" ", "Line":15, "Column":35},
	{"Type":"'<='", "Content":"<=", "Line":15, "Column":36},
	{"Type":"whitespace", "Content":" ", "Line":15, "Column":38},
	{"Type":"'&&'", "Content":"&&", "Line":15, "Column":39},
	{"Type":"whitespace", "Content":" ", "Line":15, "Column":41},
	{"Type":"'||'", "Content":"||", "Line":15, "Column":42},
	{"Type":"whitespace", "Content":" ", "Line":15, "Column":44},
	{"Type":"'&'", "Content":"&", "Line":15, "Column":45},
	{"Type":"whitespace", "Content":" ", "Line":15, "Column":46},
	{"Type":"'|'", "Content":"|", "Line":15, "Column":47},
	{"Type":"whitespace", "Content":" ", "Line":15, "Column":48},
	{"Type":"'^'", "Content":"^", "Line":15, "Column":49},
	{"Type":"whitespace", "Content":" ", "Line":15, "Column":50},
	{"Type":"'++'", "Content":"++", "Line":15, "Column":51},
	{"Type":"whitespace", "Content":" ", "Line":15, "Column":53},
	{"Type":"'--'", "Content":"--", "Line":15, "Column":54},
	{"Type":"end of line", "Content":"\r\n", "Line":15, "Column":56},
	{"Type":"'+='", "Content":"+=", "Line":16, "Column":1},
	{"Type":"whitespace", "Content":" ", "Line":16, "Column":3},
	{"Type":"'-='", "Content":"-=", "Line":16, "Column":4},
	{"Type":"whitespace", "Content":" ", "Line":16, "Column":6},
	{"Type":"'*='", "Content":"*=", "Line":16, "Column":7},
	{"Type":"whitespace", "Content":" ", "Line":16, "Column":9},
	{"Type":"'/='", "Content":"\/=", "Line":16, "Column":10},
	{"Type":"whitespace", "Content":" ", "Line":16, "Column":12},
	{"Type":"'%='", "Content":"%=", "Line":16, "Column":13},
	{"Type":"whitespace", "Content":" ", "Line":16, "Column":15},
	{"Type":"'<<='", "Content":"<<=", "Line":16, "Column":16},
	{"Type":"whitespace", "Content":" ", "Line":16, "Column":19},
	{"Type":"'>>='", "Content":">>=", "Line":16, "Column":20},
	{"Type":"whitespace", "Content":" ", "Line":16, "Column":23},
	{"Type":"'&='", "Content":"&=", "Line":16, "Column":24},
	{"Type":"whitespace", "Content":" ", "Line":16, "Column":26},
	{"Type":"'|='", "Content":"|=", "Line":16, "Column":27},
	{"Type":"whitespace", "Content":" ", "Line":16, "Column":29},
	{"Type":"'^='", "Content":"^=", "Line":16, "Column":30},
	{"Type":"end of line", "Content":"\r\n", "Line":16, "Column":32},
	{"Type":"'?'", "Content":"?", "Line":17, "Column":1},
	{"Type":"whitespace", "Content":" ", "Line":17, "Column":2},
	{"Type":"':'", "Content":":", "Line":17, "Column":3},
	{"Type":"whitespace", "Content":" ", "Line":17, "Column":4},
	{"Type":"'->'", "Content":"->", "Line":17, "Column":5},
	{"Type":"whitespace", "Content":" ", "Line":17, "Column":7},
	{"Type":"'@'", "Content":"@", "Line":17, "Column":8},
	{"Type":"whitespace", "Content":" ", "Line":17, "Column":9},
	{"Type":"'$'", "Content":"$", "Line":17, "Column":10},
	{"Type":"whitespace", "Content":" ", "Line":17, "Column":11},
	{"Type":"'#'", "Content":"#", "Line":17, "Column":12},
	{"Type":"whitespace", "Content":" ", "Line":17, "Column":13},
	{"Type":"'##'", "Content":"##", "Line":17, "Column":14},
	{"Type":"whitespace", "Content":" ", "Line":17, "Column":16},
	{"Type":"'::'", "Content":"::", "Line":17, "Column":17},
	{"Type":"end of file", "Content":"", "Line":17, "Column":19},
]
Diagnostic:[
	..\src\rfx\tests\lexer\BasicTokens.rfx(3): error 10000: illegal character 'И'
	Инвалид
	^
]
